### PR TITLE
Pods 7264

### PR DESCRIPTION
**Pull request by ngmjohnstone**

### CODE REVIEW

File: app/repositories/EventReportCacheRepository.scala [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    implicit val dateFormat: Format[DateTime] = MongoJodaFormats.dateTimeFormat
+    implicit val format: Format[JsonDataEntry] = Json.format[JsonDataEntry]
+  }
+
+  object EventReportCacheEntryFormats {
+    implicit val dateFormat: Format[DateTime] = MongoJodaFormats.dateTimeFormat
+    implicit val format: Format[EventReportCacheEntry] = Json.format[EventReportCacheEntry]
+
+    val pstrKey = "pstr"
+    val inviteePsaIdKey = "inviteePsaId"
```
Comment (grantrobson) [inline app/repositories/EventReportCacheRepository.scala:88]: Do we need this?

Comment (cyeniceri) [reply app/repositories/EventReportCacheRepository.scala:88]: I think there is no cache requirement so far for this ticket

Comment (ngmjohnstone) [reply app/repositories/EventReportCacheRepository.scala:88]: The usage of format is on 105 -> 108 as ERCacheRepo extends PlayMongoRepo, when defining domainFormat and extraCodecs.

File: conf/app.routes [hunk 2]
```diff
@@ -3,8 +3,9 @@
 GET        /overview                                controllers.EventReportController.getOverview
 GET        /versions                                controllers.EventReportController.getVersions
 
-POST       /compile-event-reporting-summary         controllers.EventReportController.compileEventReportSummary
-POST       /compile-event1-report                   controllers.EventReportController.compileEventOneReport
 POST       /submit-event-declaration-report         controllers.EventReportController.submitEventDeclarationReport
 
+POST       /save-event                              controllers.EventReportController.saveEvent
```
[the remark this answers is not in the document]
Comment (ngmjohnstone) [reply conf/app.routes:8]: Will rename and to /save and adjust Postman collection accordingly

File: app/repositories/EventReportCacheRepository.scala (hunk 1 above)
Comment (ngmjohnstone) [reply app/repositories/EventReportCacheRepository.scala:88]: The inviteePsaIdKey isn't used but the pstrKey is for indexing

Comment (grantrobson) [reply app/repositories/EventReportCacheRepository.scala:88]: yes, it was the invitee one I think needs removing

**Review by grantrobson — APPROVED** [review]

File: conf/application.conf [hunk 3]
```diff
@@ -136,6 +140,14 @@ microservice {
 mongodb {
     uri = "mongodb://localhost:27017/"${appName}
 
+    pension-scheme-event-reporting-cache {
+            maxSize = 512000
```
Comment (cyeniceri) [inline conf/application.conf:144]: Why do we have this max-size and this parent naming (pension-scheme-event-reporting-cache) ?

File: app/repositories/EventReportCacheRepository.scala [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+      Codecs.playFormatCodec(JsonDataEntry.format),
+      Codecs.playFormatCodec(DataEntry.format)
+    ),
+    indexes = Seq(
+      IndexModel(
+        Indexes.ascending(expireAtKey),
+        IndexOptions().name("dataExpiry").expireAfter(0, TimeUnit.SECONDS).background(true)
+      ),
+      IndexModel(
+        Indexes.ascending(pstrKey),
```
Comment (cyeniceri) [inline app/repositories/EventReportCacheRepository.scala:115]: why do we have index for pstrKey and apiTypesKey differently ?

**Pull request merged**